test/Microsoft.ML.Benchmarks/StochasticDualCoordinateAscentClassifierBench.cs

```diff
@@ -168,6 +170,9 @@ public void SetupPredictBenchmarks()
 
         [Benchmark]
         public void PredictIrisBatchOf5() => _trainedModel.Transform(_mlContext.Data.LoadFromEnumerable(_batches[2]));
+
+        [Benchmark]
+        public void EvaluateMetrics() => _evaluator.Evaluate(_scoredIrisTestData, DefaultColumnNames.Label, DefaultColumnNames.Score, DefaultColumnNames.PredictedLabel);
     }
 
     public class IrisData
```

Comment:
Is there a unit test which verifies the output array for the set of Top K Accuracies?

_Note: tagged to a line of code to allow for threaded discussion._

Reply:
I don't believe there's a test for checking the whole array, as that's a new feature introduced on this PR. Although we already have multiple tests testing the pre-existing TopKAccuracy member, which now uses the array behind the scenes, and those tests passed without needing a change in baselines.

If a test is needed, I think it's possible to create an IDataView with a Label column, a Scores column and a PredictedLabel column, and call the evaluator on that. I.e., I think it's not necessary to train a model from scratch to test this, instead directly creating the IDataView I mention might be preferable as in that way we could manually set the label, scores, and predicted label, so to know in advance what the correct k-accuracy should be for all k's.

Reply:
We should have test coverage for the added functionality. 

Primarily, it ensures nothing breaks it in future code changes. Secondarily, it provides an example use of the feature, and nicely, provides an example of the expected output, which makes reviewing the PR easier in the same way it helps future readers of the code.

As @antoniovs1029 mentioned it shouldn't require a new training. There's likely an existing unit test performing what Antonio specifies, so it can be expanded or duplicated.

Reply:
It seems we didn't have unit tests for Evaluate() methods (we did have tests were we called .Evaluate(), but those tests were actually testing trainers, not the .Evaluate() method itself... and we do have several tests that check evaluation metrics logic through MAML and entrypoints, but not through .Evaluate() especifically). So I've created EvaluateTests.cs:

https://github.com/dotnet/machinelearning/pull/5395/files#diff-163d9ffa622628365a9e3e316f83e72f08018acc75641db2f43ed68227458093

I've added one test, @justinormont please review and approve this PR. Thanks 😄